packages/mask/src/extension/popups/pages/Wallet/components/StartUp/index.tsx

```diff
@@ -53,50 +55,68 @@ const useStyles = makeStyles()((theme) => ({
         fontSize: 12,
         fontWeight: 400,
     },
+    placeholderDescription: {
+        fontSize: 14,
+        lineHeight: '18px',
+        fontWeight: 700,
+        color: theme.palette.maskColor.third,
+        width: 270,
+        marginTop: theme.spacing(1.5),
+        textAlign: 'center',
+    },
     mnemonicIcon: {
+        background: theme.palette.maskColor.success,
+    },
+    walletIcon: {
+        background: theme.palette.maskColor.primary,
+    },
+    iconWrapper: {
         display: 'flex',
         justifyContent: 'center',
         alignItems: 'center',
         width: 30,
         height: 30,
-        background: theme.palette.maskColor.success,
         borderRadius: '100%',
     },
 }))
 
 export const WalletStartUp = memo(() => {
     const { t } = useI18N()
-    const { classes } = useStyles()
+    const { cx, classes } = useStyles()
     const theme = useTheme()
-    const { hasPassword } = useHasPassword()
 
-    const onEnterCreateWallet = useCallback(
-        async (route: DashboardRoutes) => {
-            await browser.tabs.create({
-                active: true,
-                url: browser.runtime.getURL(
-                    `/dashboard.html#${hasPassword ? route : DashboardRoutes.CreateMaskWalletForm}`,
-                ),
-            })
-            if (Sniffings.is_firefox) {
-                window.close()
-            }
+    const [, onEnterCreateWallet] = useAsyncFn(async (route: DashboardRoutes) => {
+        const hasPassword = await WalletServiceRef.value.hasPassword()
+        await browser.tabs.create({
+            active: true,
+            url: browser.runtime.getURL(
+                `/dashboard.html#${hasPassword ? route : DashboardRoutes.CreateMaskWalletForm}${
+                    route === DashboardRoutes.RecoveryMaskWallet && !hasPassword ? '?recover=true' : ''
+                }`,
+            ),
+        })
+        if (Sniffings.is_firefox) {
+            window.close()
+        }
 
-            await Services.Helper.removePopupWindow()
-        },
-        [hasPassword],
-    )
+        await Services.Helper.removePopupWindow()
```

Comment:
Why move the firefox part to the `removePopupWindow`?

Reply:
? I didn't commit a modification here.

Reply:
IIRC we open the popup on Firefox in a different way